### PR TITLE
[Task]: Remove extracted NewsletterBundle from Core repository

### LINKED ISSUE #14802: Extract Newsletter functionality into a bundle
### Improvement description

Main task: #12094

### PR DESCRIPTION
**Pull request by mattamon**
## Changes in this pull request  
Resolves #14802 

### <samp>🤖 Generated by Copilot at f3295dd</samp>

Removed the deprecated `PimcoreNewsletterBundle` and all its related files and configurations. This bundle provided a newsletter feature that was based on an outdated library and had security and performance issues.

### <samp>🤖 Generated by Copilot at f3295dd</samp>

> _`PimcoreNewsletterBundle`_
> _Gone with the autumn wind_
> _No more security woes_

### <samp>🤖 Generated by Copilot at f3295dd</samp>

*  Remove the PimcoreNewsletterBundle and all its related files and configurations ([link](https://github.com/pimcore/pimcore/pull/14902/files?diff=unified&w=0#diff-7f05ad6d07e45c748fc59fe3f802c2fb8f3726e8a5ed6f24344f7fcbbb422a40L29-L31), F1-F18)

### CODE REVIEW

**Review by kingjia90 — CHANGES_REQUESTED** [review]
LGTM Removal is fine, just misses some useful info that could be put in this PR along the removal
- [x] Upgrade notes that is moved to `pimcore/newsletter-bundle` 
- [x] a mention in https://github.com/pimcore/pimcore/blob/11.x/doc/23_Installation_and_Upgrade/07_Updating_Pimcore/12_V10_to_V11.md about the compatibility-bridge necessity (https://github.com/pimcore/compatibility-bridge-v10/pull/5)

**Comment by mattamon** [comment]
> LGTM Removal is fine, just misses some useful info that could be put in this PR along the removal
> 
>     * [ ]  Upgrade notes that is moved to `pimcore/newsletter-bundle`
> 
>     * [ ]  a mention in https://github.com/pimcore/pimcore/blob/11.x/doc/23_Installation_and_Upgrade/07_Updating_Pimcore/12_V10_to_V11.md about the compatibility-bridge necessity ([Add Newsletter classes to add alias for old datatype namespace compatibility-bridge-v10#5](https://github.com/pimcore/compatibility-bridge-v10/pull/5))

[Newsletter] has been moved to pimcore/newsletter-bundle package.  -> Upgrade notes are already there since I put them in the Extraction PR already :)

File: doc/23_Installation_and_Upgrade/07_Updating_Pimcore/12_V10_to_V11.md [hunk 1]
```diff
@@ -27,6 +27,12 @@ If you do not use Newsletter functionality, then please make sure to run the fol
 bin/console pimcore:documents:cleanup newsletter
 ```
 
+If you use the Newsletter functionality please make sure to add the `pimcore/compatibility-bridge-v10` to your composer.json  
+```bash
+composer require pimcore/compatibility-bridge-v10
```
Comment (dvesh3) [inline doc/23_Installation_and_Upgrade/07_Updating_Pimcore/12_V10_to_V11.md]:
```suggestion
Please make sure to add the `pimcore/compatibility-bridge-v10` to your composer.json  
```bash
composer require --no-update pimcore/compatibility-bridge-v10
```
I think this should be the default step on upgrade as it provides compatibility for more classes. wdyt?

Comment (dvesh3) [reply doc/23_Installation_and_Upgrade/07_Updating_Pimcore/12_V10_to_V11.md]: and this should move to the "PRIOR THE UPDATE" section.

Comment (dvesh3) [reply doc/23_Installation_and_Upgrade/07_Updating_Pimcore/12_V10_to_V11.md]: moved the section to "Preparing for v11" guide with changes suggested above https://github.com/pimcore/pimcore/pull/14902/commits/ae34126ddd2389ab9da39e9ba79f57bf2e166ed6

**Pull request merged**